pkg/es/config/config.go

```diff
@@ -39,15 +39,26 @@ const (
 
 // IndexOptions describes the index format and rollover frequency
 type IndexOptions struct {
-	Priority          int64  `mapstructure:"priority"`
-	DateLayout        string `mapstructure:"date_layout"`
-	Shards            int64  `mapstructure:"shards"`
-	Replicas          int64  `mapstructure:"replicas"`
+	// Priority contains the of index template (ESv8 only).
+	Priority   int64  `mapstructure:"priority"`
+	DateLayout string `mapstructure:"date_layout"`
+	// Shards is the number of shards per index in Elasticsearch.
+	Shards int64 `mapstructure:"shards"`
+	// Replicas is the number of replicas per index in Elasticsearch.
+	Replicas int64 `mapstructure:"replicas"`
+	// RolloverFrequency rotates indices over the given period. For example, "day" creates
```

Comment:
Thanks for adding docs. Is this behavior unconditional or only in some cases, like when useILM is false?

Reply:
@yurishkuro I believe its unconditional. it seems to be used in the reader (https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/es/spanstore/reader.go#L275)

Reply:
no, it's not unconditional. See reader.go:151 - this param is not used when `archive || useReadWriteAliases`. Also, the doc string is not even talking about using the param at read time, it says "rotates" - and I think that's definitely not unconditional.

Reply:
@yurishkuro Is this rotation handled by ILM and this field here is just used for reading? https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/es/factory.go#L273-L282. I don't see this field being used for any writing.

Reply:
Yes, I think you were right. The index rotation is always external to Jaeger, either via a rollover script or via ILM. And the reader only needs this setting to know how to translate a [from, to] interval into distinct index names (by doing `t=t.Add(interval)`)

Reply:
But given that, the doc string is wrong as it talks about creating rollover indices.